checkov/terraform/utils.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+from typing import Optional
+
+
+class TFModule:
+    __slots__ = ("path", "name", "foreach_idx")
+
+    def __init__(self, path: str, name: str, foreach_idx: Optional[int | str] = None) -> None:
+        self.path = path
+        self.name = name
+        self.foreach_idx = foreach_idx
+
+    def __eq__(self, other) -> bool:
+        return self.path == other.path and self.name == other.name and self.foreach_idx == other.foreach_idx
+
+    def __repr__(self) -> str:
+        return f'path:{self.path}, name:{self.name}, foreach_idx:{self.foreach_idx}'
+
+
+class TFSourceModules:
+    __slots__ = "tf_source_modules"
```

Comment:
```suggestion
    __slots__ = ("tf_source_modules",)
```
I think slots needs to be an iterable